src/MSBuildTaskHost/MSBuildTaskHost.csproj

```diff
@@ -8,7 +8,7 @@
   </PropertyGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
```

Comment:
The `MSBuildTaskHost` targets .NET 3.5 so that we can host task libraries that were built against older frameworks.  I don't think we can change its target framework so @AndyGerlicher will need to weigh in

Reply:
I thought about it, but actually Concurrent collections were introduced since 4.0, so this change is necessary.

Reply:
I am not familiar with your process but you probably ship msbuild with required framework. I don't actually understand why we should care about what run msbuild? It is rather question what framework used by application, isn't it? In my optinion run-framework and target-framework are different ones.

Reply:
@jeffkl is correct. The entire purpose of the TaskHost is to target .NET 3.5. I'm not opposed to forking implementation (uhg) or adding a stub `ConcurrentQueue` class that has uses locking and is slow (perf penalty for using really old projects is probably ok). I've wanted to do the latter for `Lazy<T>` as well since I always find myself using that and having to revert it...

Reply:
The reason why I had to change `MSBuildTaskHost` too is that on Linux for some reason this code is used. If you know the way to fix it it I am okey with keeping .NET 3.5.

Reply:
MSBuildTaskHost doesn't ship on .NET Core. Perhaps on Mono? But if it does there it would be the same issue as on Windows. And it should never be used outside of legacy targeting scenarios so perf is not a primary concern there.

Reply:
Actually I can just add simple implementation of ConcurrentDictionary there, if you are okey I will fix it that way.

Reply:
Added `ConcurrentDictionary` and `ConcurrentQueue` from CoreFX